plugins/repository-hdfs/src/main/java/org/elasticsearch/repositories/hdfs/HdfsRepository.java

```diff
@@ -31,202 +42,114 @@
+        if (!Strings.hasText(uriSetting)) {
+            throw new IllegalArgumentException("No 'uri' defined for hdfs snapshot/restore");
         }
-
-        try {
-            return AccessController.doPrivileged(new PrivilegedExceptionAction<FileSystem>() {
-                @Override
-                public FileSystem run() throws IOException {
-                    return doGetFileSystem();
-                }
-            }, SecurityUtils.AccBridge.acc());
-        } catch (PrivilegedActionException pae) {
-            Throwable th = pae.getCause();
-            if (th instanceof Error) {
-                throw (Error) th;
-            }
-            if (th instanceof RuntimeException) {
-                throw (RuntimeException) th;
-            }
-            if (th instanceof IOException) {
-                throw (IOException) th;
-            }
-            throw new ElasticsearchException(pae);
+        URI uri = URI.create(uriSetting);
+        if (!"hdfs".equalsIgnoreCase(uri.getScheme())) {
+            throw new IllegalArgumentException(
+                    String.format(Locale.ROOT, "Invalid scheme [%s] specified in uri [%s]; only 'hdfs' uri allowed for hdfs snapshot/restore", uri.getScheme(), uriSetting));
+        }
+        if (Strings.hasLength(uri.getPath()) && uri.getPath().equals("/") == false) {
+            throw new IllegalArgumentException(String.format(Locale.ROOT,
+                    "Use 'path' option to specify a path [%s], not the uri [%s] for hdfs snapshot/restore", uri.getPath(), uriSetting));
         }
-    }
 
-    private FileSystem doGetFileSystem() throws IOException {
-        // check if the fs is still alive
-        // make a cheap call that triggers little to no security checks
-        if (fs != null) {
-            try {
-                fs.isFile(fs.getWorkingDirectory());
-            } catch (IOException ex) {
-                if (ex.getMessage().contains("Filesystem closed")) {
-                    fs = null;
-                }
-                else {
-                    throw ex;
-                }
-            }
+        String pathSetting = repositorySettings.settings().get("path");
+        // get configuration
+        if (pathSetting == null) {
+            throw new IllegalArgumentException("No 'path' defined for hdfs snapshot/restore");
         }
-        if (fs == null) {
-            Thread th = Thread.currentThread();
-            ClassLoader oldCL = th.getContextClassLoader();
-            try {
-                th.setContextClassLoader(getClass().getClassLoader());
-                return initFileSystem(repositorySettings);
-            } catch (IOException ex) {
-                throw ex;
-            } finally {
-                th.setContextClassLoader(oldCL);
+        
+        int bufferSize = (int) repositorySettings.settings().getAsBytesSize("buffer_size", new ByteSizeValue(100, ByteSizeUnit.KB)).bytes();
```

Comment:
OK, I actually have no clue what this setting is, or what it is doing: its not new in this PR, its just the line of code is moved from somewhere else...